plotly/plotly/plotly.py

```diff
@@ -1216,6 +1218,31 @@ def validate_credentials(credentials):
         raise exceptions.PlotlyLocalCredentialsError()
 
 
+def secret_key(r, **payload):
+    """
+    update plot's url to include the scret key
+
+    """
+
+    plot_id = r['url'].replace("https://plot.ly/~" + payload['un'] + "/", "")
+    url = "https://api.plot.ly/v2/files/" + payload['un'] + ":" + plot_id
+    plot_kwargs = json.loads(payload['kwargs'])
+    response = requests.patch(url,
```

Comment:
This is OK for now, but when we move to wrapping the new `/plots` endpoint, this will just be a single api call. We *could* update `clientresp` to handle this now, but we're transitioning away from clientresp so I'd rather not add anything new to it.